assets/opstools/BuildApp/controllers/data_fields/date.js

```diff
@@ -2,66 +2,472 @@ steal(function () {
 	var componentIds = {
 		editView: 'ab-new-date',
 		includeTime: 'ab-new-date-include-time',
+		datedisPlay: 'ab-new-date-date-display',
+		
+		includedayFormat : 'ab-new-date-includedayFormat',
+		includemonthFormat : 'ab-new-date-includemonthFormat',
+		includeyearFormat : 'ab-new-date-includeyearFormat',
+		
+		typedayformatDelimiters  : 'ab-new-date-typedayformatDatetime',
+		typemonthformatDelimiters  : 'ab-new-date-typemonthdayformatDatetime',
+		typeyearformatDelimiters  : 'ab-new-date-typeyearformatDatetime',
+		
+		includedayOrder  : 'ab-new-date-includedayOrder',
+		includemonthOrder  : 'ab-new-date-includemonthOrder',
+		includeyearOrder  : 'ab-new-date-includeyearOrder',
+		
 	};
 
 	// General settings
 	var dateDataField = {
 		name: 'date',
 		type: ['datetime', 'date'], // http://sailsjs.org/documentation/concepts/models-and-orm/attributes#?attribute-options
 		icon: 'calendar',
-		menuName: AD.lang.label.getLabel('ab.dataField.date.menuName') || 'Date',
+		menuName: 'Date',
 		includeHeader: true,
 		description: ''
 	};
+	
+	function getDelimiters(d){
+		switch (d) {
+				case 'comma':
+					return ",";
+					break;
+				case 'slash':
+					return "/";
+					break;
+				case 'space':
+					return " ";
+					break;
+				case 'dash':
+					return "-";
+					break;
+			}
+	}
+	
+	function setformatdatetimeOrder(orderday,ordermonth,orderyear,includedayformat,includemonthformat,includeyearformat,
```

Comment:
please use camel case when creating function and variable names.

so: setFormatDateTimeOrder(  orderDay, orderMonth, orderYear,  ....